internal/pkg/runtime/launcher/oci/cdi_linux.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2018-2022, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// Includes code from https://github.com/containers/podman
+// Released under the Apache License Version 2.0
+
+package oci
+
+import (
+	"fmt"
+
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+// addCDIDevice adds an array of CDI devices to an existing spec.
+func addCDIDevice(spec *specs.Spec, cdiDevices []string) error {
+	// Get the CDI registry, passing a cdi.WithAutoRefresh(false) option so that CDI registry files are not scanned asynchronously. (We are about to call a manual refresh, below.)
+	registry := cdi.GetRegistry(cdi.WithAutoRefresh(false))
+
+	// Refresh the CDI registry.
+	if err := registry.Refresh(); err != nil {
+		return fmt.Errorf("Error encountered refreshing the CDI registry: %v", err)
+	}
+
+	for _, cdiDevice := range cdiDevices {
```

Comment:
Note that the `registry.InjectDevices` function accepts a variadic list of fully-qualified device names. This includes a check for a valid name. As such I don't know whether introducing an `addSingleCDIDevice` function adds specific value.